internal/app/machined/pkg/controllers/k8s/node_status.go

```diff
@@ -91,6 +91,8 @@ func (ctrl *NodeStatusController) Run(ctx context.Context, r controller.Runtime,
 			kubernetesClient = nil
 		}
 
+		watchErrors = 0
+		watchReady = false
```

Comment:
I think you can remove `watchErrors = 0` below now after calling `closeWatcher()`.